overlay/flask_server.py

```diff
@@ -40,63 +39,45 @@ def default_route():
     if validation_request_str == '':
         return 'Empty request submitted', HTTPStatus.BAD_REQUEST
 
-    validation_request = json.loads(validation_request_str)
+    validation_request: dict = json.loads(validation_request_str)
     pprint(validation_request)
 
     # Check if the POST request has the file part
     if 'file' not in request.files:
         return 'No file included in request', HTTPStatus.BAD_REQUEST
 
-    file = request.files['file']
+    file: FileStorage = request.files['file']
 
     # If the user does not select a file, the browser submits an
     # empty file without a filename.
     if file.filename == '':
         return 'Empty file submitted', HTTPStatus.BAD_REQUEST
 
     if file and allowed_file(file.filename):
-        file_bytes = file.read()
+        file_bytes: bytes = file.read()
 
         hasher = hashlib.md5()
         hasher.update(file_bytes)
-        md5_hash = hasher.hexdigest()
+        md5_hash: str = hasher.hexdigest()
         info(f"Uploaded file of size {len(file_bytes)} bytes, and hash: {md5_hash}")
 
         with grpc.insecure_channel(f"{app.config['MASTER_HOSTNAME']}:{app.config['MASTER_PORT']}") as channel:
-            stub = validation_pb2_grpc.MasterStub(channel)
-            model_file = ModelFile(
-                type="zip",
-                md5_hash=md5_hash,
-                data=file_bytes
-            )
-
-            validation_grpc_request = ValidationJobRequest(
-                job_mode=validation_request["job_mode"],
-                model_framework=validation_request["model_framework"],
-                model_category=validation_request["model_category"],
-                mongo_host=validation_request["mongo_host"],
-                mongo_port=validation_request["mongo_port"],
-                read_config=MongoReadConfig(
-                    read_preference=validation_request["read_config"]["read_preference"],
-                    read_concern=validation_request["read_config"]["read_concern"]
-                ),
-                database=validation_request["database"],
-                collection=validation_request["collection"],
-                label_field=validation_request["label_field"],
-                feature_fields=validation_request["feature_fields"],
-                normalize_inputs=validation_request["normalize_inputs"],
-                limit=validation_request["limit"],
-                sample_rate=validation_request["sample_rate"],
-                validation_metric=validation_request["validation_metric"],
-                gis_joins=validation_request["gis_joins"],
-                model_file=model_file
-            )
-
-            validation_grpc_response = stub.SubmitValidationJob(validation_grpc_request)
+            stub: validation_pb2_grpc.MasterStub = validation_pb2_grpc.MasterStub(channel)
+
+            # Build and log gRPC request
+            validation_grpc_request: ValidationJobRequest = Parse(validation_request_str, ValidationJobRequest())
```

Comment:
This does the job of what all the lines in the red do above, but the `model_file` still needs to be set below.